src/snowflake/snowpark/session.py

```diff
@@ -1924,11 +1925,39 @@ def table_function(
             - :meth:`Session.generator`, which is used to instantiate a :class:`DataFrame` using Generator table function.
                 Generator functions are not supported with :meth:`Session.table_function`.
         """
+        # AST.
+        stmt = self._ast_batch.assign()
+        expr = with_src_position(stmt.expr.apply_expr)
+        if isinstance(func_name, TableFunctionCall):
+            expr.fn.udtf.name = func_name.name
+            for arg in func.arguments:
+                build_const_from_python_val(arg, expr.pos_args.add())
+            for k in func.named_arguments:
+                entry = expr.named_args.add()
+                entry._1 = k
+                build_const_from_python_val(func.named_arguments[k], entry._2)
```

Comment:
This block appears identical on the two branches. Put `func.arguments` and `func.named_arguments` in a couple of variables and do this step unconditionally?